csharp/src/Ice/OutputStream.cs

```diff
@@ -1151,10 +1193,10 @@ private int Distance(Position start)
         /// <summary>Computes the amount of data written from the start position to the current position and writes that
         /// value at the start position (as a fixed-length size).</summary>
         /// <param name="start">The start position.</param>
-        private void EndSize(Position start)
+        private void EndFixedLengthSize(Position start)
         {
             Debug.Assert(start.Offset >= 0);
-            RewriteInt(Distance(start) - 4, start);
+            RewriteFixedLengthSize(Distance(start) - 4, start);
```

Comment:
WriteFixedLengthSize checks if size can be out of range but AFAICT the implementation of Distance doesn't check for overflows... Shouldn't it check overflows or otherwise we could end writing a bogus size here?

Reply:
Added check for overflows in Distance.